### ISSUE TITLE
Coupon redemption limit = 0 doesn't work

### ISSUE THREAD

**Issue by phthhieu** [fnando/coupons#16]
1. I create a coupon with Redemption Limit = 0 (so the number of time usage should be ∞)
2. Then I use this coupon by this code:
   Coupons.redeem(coupon, amount: total_amount_dollars)
3. From now, if I check coupon with:
   Coupons.apply(coupon, amount: total_amount_dollars)

Real result: The discount amount will be returned 0 in step 3
Expectation: it shouldn't be 0, because Redemption Limit = 0

Could you guys help on this? Thanks so much

**Comment by cabgfx** [comment]
This is working for me, as you've described it here.
I have had it running in production for some time now, and – AFAIK – there hasn't been updates over the past couple months, so I suspect it might be related to how you've set up the coupon.

Could you post a screenshot of your Edit page, so we can see how you've set up the coupon?

**Comment by kranthi1027** [comment]
Yes. Limit when set to 0 is not taking it as infinity. I've tried it.

**Comment by cabgfx** [comment]
Actually, I misunderstood this in my first reply.

0 for redemption limit does not work for unlimited redemptions. In fact, it only allows 1 redemption.

Any pointers on this, @fnando ?

**Comment by facundofarias** [comment]
Is this still open? :/
